application/helpers/common_helper.php

```diff
@@ -260,7 +260,7 @@ function convertGETtoPOST($url)
     $stack = explode('?', $url);
     $calledscript = array_shift($stack);
     $query = array_shift($stack);
-    $aqueryitems = explode('&', $query);
+    $aqueryitems = explode('&', (string) $query);
```

Comment:
I know it's more a rewrite, but i thni it's better to check `?` here https://github.com/LimeSurvey/LimeSurvey/blob/f7ab872d576308c2cb65d213f9d01633b70f0a3d/application/helpers/common_helper.php#L259

And return directly the url with no POST value.

Reply:
Not sure what you mean. Can you expand a bit more?
Still, I would try to stick with just the needed updates. Other enhancements could be done on different tickets

Reply:
Something like this

```
function convertGETtoPOST($url)
{
    $url = preg_replace('/&amp;/i', '&', (string) $url);
    if (strpos('?', $url) === false {
        return "window.LS.sendPost(\"" . $url ."\");";
    }
    $stack = explode('?', $url);
```

Reply:
I would do afer hacing an automated test for it. Not now. For me, it doesn't worth the risk of updating it now.